test/compilable/commontype.d

```diff
@@ -0,0 +1,491 @@
+// REQUIRED_ARGS: -mcpu=avx2
```

Comment:
A comment explaining the purpose of these tests would be nice.

Reply:
The purpose is tweaking `typeMerge()` without unintentionally breaking something.